app/src/main/java/br/com/dionataferraz/vendas/HomeActivity.kt

```diff
@@ -1,18 +1,47 @@
 package br.com.dionataferraz.vendas
 
-import androidx.appcompat.app.AppCompatActivity
+import android.content.Intent
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
 import br.com.dionataferraz.vendas.databinding.ActivityHomeBinding
+import br.com.dionataferraz.vendas.transactions.ModifyBalanceActivity
+import br.com.dionataferraz.vendas.transactions.TransactionsActivity
 
 class HomeActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityHomeBinding
+    private lateinit var viewModel: HomeViewModel
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        ActivityHomeBinding.inflate(layoutInflater).run {
-            binding = this
-            setContentView(root)
+
+        binding = ActivityHomeBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        viewModel = HomeViewModel()
+
+        viewModel.getBalance()
+
+        viewModel.showBalance.observe(this) { balance ->
+            binding.tvAccountBalance.text = balance.toString()
+
+        }
+
+        binding.tvAccountBalance.setOnClickListener {
+            val intent = Intent(App.context, TransactionsActivity::class.java)
```

Comment:
Como estamos em uma activity não precisamos acessar o `App.context`, podemos acessar o context direto pelo `this`
```suggestion
            val intent = Intent(this, TransactionsActivity::class.java)
```